component/async/component_test.go

```diff
@@ -431,18 +469,27 @@ func (mm *mockMessage) Payload() []byte {
 type mockProcessor struct {
 	errReturn bool
 	execs     int
+	mux       sync.Mutex
```

Comment:
Minor: A convention is to have the mutex before the fields that it protects.

Reply:
thanks, will apply in next commit